examples/bri-3/src/bri/zkp/agents/proofStorage.agent.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
+import { BlockchainService } from '../components/blockchain/blockchain.service';
+
+import { Proof } from '../models/proof';
+
+@Injectable()
+export class ProofStorageAgent {
+  constructor(private storageAgent: BlockchainService) {}
+
+  async storeProofInShieldContract(proof: Proof): Promise<void> {
```

Comment:
@biscuitdey Perhaps we should agree on using either "ShieldContract" or "OnChain" and stick with one or the other per your conversation with @Therecanbeonlyone1969.